src/mlpack/methods/ann/activation_functions/softplus_function.hpp

```diff
@@ -51,9 +51,21 @@ class SoftplusFunction
    */
   static double Fn(const double x)
   {
+    const double val = std::log(1 + std::exp(x));
     if (x < DBL_MAX)
-      return x > -DBL_MAX ? std::log(1 + std::exp(x)) : 0;
-    return 1.0;
+    {
+      if (x > -DBL_MAX)
+      {
+        if (std::isfinite(val))
+          return val;
+        else
+          return x;
+      }
+      else
+        return 0;
+    }
+    else
+      return 1.0;
```

Comment:
Why should the value of `log(1 + exp(x))` be `1` when the input `x` is a big integer?

Reply:
I too was wondering about that. Thought maybe it's made in use somewhere in the code, Ill change and see if any test fails..

Reply:
@zoq Since you were involved with this code, maybe you could provide a rationale for this?

Reply:
There is no rational explanation, it's not correct.